spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -400,6 +400,21 @@ class DeltaAnalysis(session: SparkSession)
           throw DeltaErrors.notADeltaTableException("RESTORE")
       }
 
+    // Resolve as a resolved table if the path is for delta table. For non delta table, we keep the
+    // path and pass it along.
+    case u: UnresolvedPath =>
+      val table = getPathBasedDeltaTable(u.path)
+      if (!table.tableExists) {
+        u
+      } else {
+        val catalog = session.sessionState.catalogManager.currentCatalog.asTableCatalog
```

Comment:
What if the current catalog is not the target catalog in the table identifier?
E.g. current catalog is hive, but the query is selecting from `customer_catalog.db.table1`

Reply:
I fixed mine in https://github.com/delta-io/delta/pull/2012/commits/5f1b9b81e4d471d0a593c92bcb4d4855948574d9

Reply:
This is fixed now.